src/legacy/core_plugins/kibana/public/discover/embeddable/search_embeddable_factory.ts

```diff
@@ -46,35 +53,64 @@ export class SearchEmbeddableFactory extends EmbeddableFactory {
     });
   }
 
-  public getEditPath(panelId: string) {
-    return this.searchLoader.urlFor(panelId);
+  public isEditable() {
+    return capabilities.get().discover.save as boolean;
+  }
+
+  public canCreateNew() {
+    return false;
+  }
+
+  public getDisplayName() {
+    return i18n.translate('kbn.embeddable.search.displayName', {
+      defaultMessage: 'search',
+    });
   }
 
-  /**
-   *
-   * @param {Object} panelMetadata. Currently just passing in panelState but it's more than we need, so we should
-   * decouple this to only include data given to us from the embeddable when it's added to the dashboard. Generally
-   * will be just the object id, but could be anything depending on the plugin.
-   * @param onEmbeddableStateChanged
-   * @return {Promise.<Embeddable>}
-   */
-  public create(
-    { id }: EmbeddableInstanceConfiguration,
-    onEmbeddableStateChanged: OnEmbeddableStateChanged
-  ) {
-    const editUrl = this.getEditPath(id);
-    const editable = capabilities.get().discover.save as boolean;
+  public async createFromSavedObject(
```

Comment:
we use await earlier... I'm going to actually just  convert this to using await. it's possible whatever required this before doesn't anymore because of all the changes and there are no errors and no UI issues (that I notice) when I change it over.